### DISCUSSION TITLE
Set value variable from Textfield

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#542] [General]
how to set variable value in Textfield
example in flutter
```
Textfield(
controller: first_name,
........
)
```
how to in Flet
example
```
name = ""
Textfield(
value=name
...
)
```
is not work

**Comment by ndonkoHenri** [comment]
Have a look at examples from the docs: https://flet.dev/docs/controls/textfield

If you don't still understand how it works, then provide the full not-working code sample, so we could understand the issue and help you out. 🙂

**Reply by bobwatcherx** [reply]
> Have a look at examples from the docs: https://flet.dev/docs/controls/textfield
> 
> If you don't still understand how it works, then provide the full not-working code sample, so we could understand the issue and help you out. slightly_smiling_face

this code is not work
```
import flet
from flet import *
from flet import colors

def main(page:Page):
	name = ""
	page.add(
			Textfield(
			value=name
			label="new name"
			)

		)

flet.app(target=main,)
```
error log is
```
SyntaxError: invalid syntax. Perhaps you forgot a comma?
  File "/home/ubuntux/belajar/fletapp/app2/main.py", line 9
    value=name
          ^^^^
```

**i wan to set value textfield is name**

**Reply by ndonkoHenri** [reply]
It is stated clearly on your console log: **you forgot a comma**. Just after `value=name`.

```python
Textfield(
	value=name ,
	label="new name"
)
```

Let me know if it works now.

**Comment by bobwatcherx** [comment]
its work ,
 but i have problem . why listview canot append from string 
here example code :
```
import flet
from flet import TextField,Page,Column,ListView

def main(page:Page):
	name = ""
	lv = ListView(auto_scroll=True,padding=10)


	def addtask(e):
		lv.controls.append(name)
		print(lv)
		page.update()


	page.add(
		Column([
			TextField(value=name,label="nice",
				on_submit=addtask
				),
			lv
			])


		)
flet.app(target=main)
```
and the script below also doesn't work
```
import flet
from flet import TextField,Page,Column,ListView

def main(page:Page):
	lv = ListView(auto_scroll=True,padding=10)

	def addtask(e):
		lv.controls.append(txt_name.value)
		print(lv)
		page.update()

	txt_name = TextField(label="bego",on_submit=addtask)

	page.add(
		Column([
			txt_name,
			lv
			])


		)
flet.app(target=main)
```
error logs
AttributeError: 'str' object has no attribute '_build_add_commands'


i want to push textfield value to listview and display it to screen under textfield widget
thanks

**Reply by ndonkoHenri** [reply]
`ListView` takes only `Control`s. You are trying to pass/force in a string, and hence got that error.

You have to wrap/put your string value in a `Text` control. 👇 

`lv.controls.append(Text(txt_name.value))`

Let me know if it works now.

**Comment by bobwatcherx** [comment]
yes work thx
now , **i wan to add button edit and delete** beside listview
can you give example 

```
import flet
from flet import *
from flet import colors

def main(page:Page):
	page.title = "goblok"
	name = ""
	listdata = ["dqwdqw"]
	
	txt_name = TextField(
	label="add task",
	)

	lv = ListView(expand=1,spacing=1,padding=20,auto_scroll=True)

	def addtask(e):
		lv.controls.append(Text(txt_name.value))
		print(listdata)
		page.update()

	
	page.add(
		AppBar(title=Text("niceede",color=colors.RED,weight="w900",),
			bgcolor=colors.BLUE),
		Column(
			[
			Row([
				ElevatedButton(
				on_click=addtask,
				text="add",
				),
				txt_name,
				],
				alignment="center"
				),
			lv
			]

			)

		)
flet.app(target=main,)
```
![gaaaaaoblok_001](https://user-images.githubusercontent.com/52072811/199687493-258964e7-b07e-4f91-b77b-b41182cf39b6.png)

**Reply by ndonkoHenri** [reply]
Please follow this quick tutorial from the docs on how to Build a ToDo app with Flet.

https://flet.dev/docs/tutorials/python-todo

If after that you still have some issues or stuffs you don't understand, you could then [open a discussion](https://github.com/flet-dev/flet/discussions), so anyone can help you.

**Comment by bobwatcherx** [comment]
is property control in listview

for example like this 
Listview(control=Array)

so as not to make it like constanta
like this
lv = ListView()

**Reply by ndonkoHenri** [reply]
Your question is not really clear. But,
ListView has a property named [`controls`](https://flet.dev/docs/controls/listview#controls) which is a list of the controls you wish to add into the ListView.

If you wish, you could add the ListView to a variable like this:
`lv=ListView()`

Or use Refs: https://flet.dev/blog/control-refs/

Hope I answered your question.
